chore(KNO-13495): Add broadcast email best practices (#1472)
knocklabs/docs: content/concepts/broadcasts.mdx

Broadcasts are a way to power one-time, cross-channel messaging to your users through Knock. They are built on Knock's [workflow engine](/concepts/workflows), giving you the power to create intelligent messaging across all of the [channels you've configured in Knock](/concepts/channels).

<Callout
  type="info"
  title="Consider using a separate email channel for marketing broadcasts."

>   text="If you plan to use broadcasts for high-volume or frequent marketing sends, we recommend configuring a dedicated email channel that is separate from your transactional email channel, and using a separate sending domain or subdomain where possible. Marketing and transactional email can have different engagement and complaint patterns, and separating them helps protect the deliverability and sender reputation of your transactional email."

Reviewer: ```suggestion
  text="If you plan to send frequent or high-volume marketing broadcasts, we recommend configuring a dedicated email channel that is separate from your transactional email channel, with a unique sending domain or subdomain where possible. Marketing and transactional email can have different engagement and complaint patterns, and separating them helps protect the deliverability and sender reputation of your transactional email."
```
